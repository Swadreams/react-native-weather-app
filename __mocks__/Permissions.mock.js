export const RESULTS = {
  UNAVAILABLE: 'unavailable',
  DENIED: 'denied',
  BLOCKED: 'blocked',
  GRANTED: 'granted',
};

export const openSettings = jest.fn(async () => {});

export const check = jest.fn(() => {
  return Promise.resolve(RESULTS.GRANTED);
});

export const request = jest.fn(() => {
  return Promise.resolve(RESULTS.GRANTED);
});

export const PERMISSIONS = {ANDROID: {}, IOS: {}};
