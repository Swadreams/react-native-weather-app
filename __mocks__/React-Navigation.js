const MockNavigation = {
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  goBack: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  setParams: jest.fn(),
  dispatch: jest.fn(),
  isFocused: jest.fn().mockReturnValue(true),
  canGoBack: jest.fn().mockReturnValue(true),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dangerouslyGetState: jest.fn(),
  setOptions: jest.fn(),
};

const ReactNavigationNative = jest.requireActual('@react-navigation/native');

const MockNativeNavigation = {
  ...ReactNavigationNative,
  useLinking: jest.fn(),
  useNavigation: jest.fn().mockReturnValue(MockNavigation),
  useFocusEffect: jest.fn().mockImplementation(callback => {
    const unsubscribe = callback();
    if (typeof unsubscribe === typeof Function) {
      unsubscribe();
    }
    return unsubscribe;
  }),
  useIsFocused: jest.fn().mockReturnValue(true),
  DrawerActions: {
    toggleDrawer: jest.fn(),
  },
};

const {createStackNavigator} = jest.requireActual('@react-navigation/stack');

export const MockStackNavigation = {
  createStackNavigator,
  CardStyleInterpolators: {},
};

export {MockNavigation, MockNativeNavigation};
