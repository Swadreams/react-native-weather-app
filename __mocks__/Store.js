import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];

const initialState = {
  weather: {
    locations: [],
    currentLocation: {},
    currentWeather: {},
    currentWeatherLocation: {},
    historicalWeather: {},
    homeWeather: {},
    homeLocation: {},
    homeWeatherLocation: {},
    isLoading: {},
  },
};

let store;

export const configureMockStore = data => {
  const initialData = data ? {...initialState, ...data} : initialState;
  store = configureStore(middlewares)(initialData);
  return store;
};

export const getMockStore = () => {
  return store;
};
