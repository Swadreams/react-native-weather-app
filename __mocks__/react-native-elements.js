jest.mock('react-native-elements', () => {
  const SearchBar = {
    placeholder: '',
    onChangeText: jest.fn(),
    value: jest.fn(),
    lightTheme: jest.fn(),
  };

  return {
    ...SearchBar,
  };
});
