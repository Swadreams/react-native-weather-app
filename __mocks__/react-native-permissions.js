import * as MockPermissions from './Permissions.mock';
jest.mock('react-native-permissions', () => MockPermissions);
