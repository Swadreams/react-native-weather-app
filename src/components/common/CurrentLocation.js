import React, {useEffect} from 'react';
import {StyleSheet, View} from 'react-native';
import Geolocation from '@react-native-community/geolocation';
import {useDispatch, useSelector} from 'react-redux';
import {
  dispatchFetchCurrentWeather,
  dispatchSetHomeLocation,
} from '../../actions/weatherActions';
import {WeatherDetails} from './weatherDetails';
import {isEmptyObject} from '../../shared/helper';

export const CurrentLocation = () => {
  const dispatch = useDispatch();
  const homeLocation = useSelector(state => state.weather.homeLocation);
  const homeWeather = useSelector(state => state.weather.homeWeather);

  const setCurrentLocarion = () => {
    Geolocation.getCurrentPosition(
      position => {
        const {latitude, longitude} = position.coords;
        dispatch(dispatchSetHomeLocation({lat: latitude, lon: longitude}));
        dispatch(dispatchFetchCurrentWeather(true));
      },
      () => {
        console.log(false);
      },
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000},
    );
  };

  useEffect(() => {
    if (isEmptyObject(homeLocation) || isEmptyObject(homeWeather)) {
      setCurrentLocarion();
    }
  }, []);

  return (
    <View style={styles.container}>
      {!isEmptyObject(homeLocation) && !isEmptyObject(homeWeather) && (
        <WeatherDetails location={homeLocation} weather={homeWeather} />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  message: {
    fontSize: 18,
    color: '#ffffff',
  },
});
