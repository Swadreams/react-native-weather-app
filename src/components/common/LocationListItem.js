import {useNavigation} from '@react-navigation/native';
import React from 'react';
import {Pressable} from 'react-native';
import {ListItem} from 'react-native-elements';
import {useDispatch} from 'react-redux';
import {dispatchSetCurrentLocation} from '../../actions/weatherActions';

export const LocationListItem = ({location}) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const navigateToDetails = locDetails => {
    dispatch(dispatchSetCurrentLocation(locDetails));
    navigation.navigate('Details');
  };

  return (
    <Pressable onPress={() => navigateToDetails(location)}>
      <ListItem bottomDivider>
        <ListItem.Content>
          <ListItem.Title>
            {location.name}, {location.region}, {location.country}
          </ListItem.Title>
        </ListItem.Content>
        <ListItem.Chevron />
      </ListItem>
    </Pressable>
  );
};
