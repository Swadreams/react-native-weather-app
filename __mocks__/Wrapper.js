import React from 'react';
import {Provider} from 'react-redux';

import {MockNavigation} from './React-Navigation';
import {getMockStore} from './Store';
import {Locations} from '../src/components/screens/LandingPage';
import {Geolocation} from './@react-native-community/geolocation';
import {PERMISSIONS, request} from './react-native-permissions';

const MockWrapper = ({component, initialRoute}) => {
  const screenName = 'Locations';
  const store = getMockStore();

  const ComponentWithNavProps = () => {
    return (
      <component
        route={{...initialRoute, key: screenName, name: Locations}}
        navigation={MockNavigation}
      />
    );
  };

  return store ? (
    <Provider store={store}>
      <ComponentWithNavProps />
    </Provider>
  ) : (
    <ComponentWithNavProps />
  );
};

MockWrapper.defaultProps = {
  initialRoute: {},
};

export {MockWrapper};
