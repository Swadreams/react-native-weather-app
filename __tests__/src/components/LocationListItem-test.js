import 'react-native';
import React from 'react';
import {LocationListItem} from '../../../src/components/common/LocationListItem';
import {create} from 'react-test-renderer';
import {MockWrapper} from '../../../__mocks__/Wrapper';

describe('LocationListItem Component', () => {
  let component;
  beforeEach(() => {
    component = create(<MockWrapper component={LocationListItem} />);
  });

  it('Message snapshot ', () => {
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
