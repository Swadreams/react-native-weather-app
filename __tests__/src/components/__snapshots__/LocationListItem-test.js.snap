// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationListItem Component Message snapshot  1`] = `
<component
  navigation={
    Object {
      "addListener": [MockFunction],
      "canGoBack": [MockFunction],
      "dangerouslyGetParent": [MockFunction],
      "dangerouslyGetState": [MockFunction],
      "dispatch": [MockFunction],
      "goBack": [MockFunction],
      "isFocused": [MockFunction],
      "navigate": [MockFunction],
      "pop": [MockFunction],
      "popToTop": [MockFunction],
      "push": [MockFunction],
      "removeListener": [MockFunction],
      "replace": [MockFunction],
      "reset": [MockFunction],
      "setOptions": [MockFunction],
      "setParams": [MockFunction],
    }
  }
  route={
    Object {
      "key": "Locations",
      "name": undefined,
    }
  }
/>
`;
